lib/ads.js

```diff
@@ -510,9 +521,75 @@ var getSymbols = function (cb) {
   })
 }
 
+var getHandles = function (commandOptions, cb) {
+  var ads = this
+
+  var data = commandOptions.reduce(function (result, handle) {
+    return result + handle.symname
+  }, '')
+
+  var buf = Buffer.alloc(commandOptions.length * 16 + data.length)
+
+  commandOptions.forEach(function (handle, index) {
+    handle = parseHandle(handle)
+
+    buf.writeUInt32LE(ADSIGRP.GET_SYMHANDLE_BYNAME, index * 16 + 0) // indexGroup
+    buf.writeUInt32LE(0x00000000, index * 16 + 4) // indexOffset
+    buf.writeUInt32LE(4, index * 16 + 8) // handleSize
+    buf.writeUInt32LE(handle.symname.length, index * 16 + 12) // stringLength
+  })
+
+  buf.write(data, (commandOptions.length) * 16 + 0)
+
+  var request = {
+    indexGroup: ADSIGRP.SUMUP_READWRITE,
+    indexOffset: commandOptions.length,
+    writeBuffer: buf,
+    readLength: commandOptions.length * 12,
+    symname: 'getHandles'
+  }
+
+  writeReadCommand.call(ads, request, function (err, result) {
+    if (err) {
+      cb.call(ads, err)
+    } else {
+      if (result.length > 0) {
+        var resultpos = 0
+        var handlespos = commandOptions.length * 8
+
+        commandOptions.forEach(function (handle) {
+          if (handle.symname !== undefined) {
+            var adsError = result.readUInt32LE(resultpos)
+            resultpos += 4
+
+            handle.err = getError(adsError)
```

Comment:
if we export the "const errors = {" structure, we could just write the error code here. This would save memory for many errors. but we should then describe in readme.md how to evaluate it.

Reply:
Good point. I would like to move and clean some things today. The whole debugging parts can be also moved into another file.

Reply:
let's do one step to the other. I think we can do that things and other cleanups and reorganizations in another PR.